Gems/Atom/RHI/Metal/Code/Source/Platform/iOS/RHI/Metal_RHI_iOS.cpp

```diff
@@ -29,13 +29,20 @@ namespace Platform
         return physicalDeviceList;
     }
 
-    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval)
+    float GetRefreshRate()
     {
+        NativeScreenType* nativeScreen = [NativeScreenType mainScreen];
+        return [nativeScreen maximumFramesPerSecond];
+    }
+
+    void PresentInternal(id <MTLCommandBuffer> mtlCommandBuffer, id<CAMetalDrawable> drawable, float syncInterval, float refreshRate)
+    {
+        float presentAfterMinimumDuration = syncInterval / refreshRate;
         bool hasPresentAfterMinimumDuration = [drawable respondsToSelector:@selector(presentAfterMinimumDuration:)];
                 
-        if (hasPresentAfterMinimumDuration && syncInterval > 0.0f)
+        if (hasPresentAfterMinimumDuration && hasPresentAfterMinimumDuration > 0.0f)
         {
-            [mtlCommandBuffer presentDrawable:drawable afterMinimumDuration:syncInterval];
+            [mtlCommandBuffer presentDrawable:drawable afterMinimumDuration:presentAfterMinimumDuration];
```

Comment:
Do we need a guard to ensure the iOS version is 10.3+? From apple [docs](https://developer.apple.com/documentation/metal/mtlcommandbuffer/2806849-presentdrawable?language=objc)

Reply:
O3DE min spec on ios is iOS 13 so this should be safe.